src/features/staging.ts

```diff
@@ -84,7 +84,7 @@ export const StagingMixin = (ModelViewerElement:
         [$tick](time: number, delta: number) {
           super[$tick](time, delta);
 
-          if (this.autoRotate) {
+          if (this.autoRotate && (this as any)[$scene].model.hasModel()) {
```

Comment:
The assumption here is that the base model only has children in it when the gltF is loaded and the poster is dismissed( or if there was no poster to begin with). That's how it seems to be working, with model.hasModel() only being true once `setSource` is called after the poster is dismissed. If this no longer becomes the case, this fix might not work.

I also don't know if `[$scene].model.hasModel()`will incur a performance cost being called on each tick. Right now hasModel calls `!!this.modelContainer.children.length;`

Reply:
If I understand the issue correctly, the psuedocode for the correct logic should be something along the lines of:

```javascript
if (autoRotate AND modelIsLoaded AND posterIsHidden) {
  rotateModel();
}
```

The clause `modelIsLoaded AND posterIsHidden` could be satisfied by a common notion of `modelIsVisible`, which would let us know that the model has been loaded, rendered and is actually being shown to the viewer.

What if we add an accessor `get modelIsVisible() { ... }` in the base class that can be overridden by mixins (like `LoadingMixin`) to specialize the definition? Then the conditional logic in `StagingMixin` can be:

```javascript
if (this.autoRotate && this.modelIsVisible) {
  // ...
}
```

And  this way `StagingMixin` doesn't need to worry about the implementation details of `LoadingMixin`. I think this would also excuse you from testing the explicit integration of `StagingMixin` and `LoadingMixin` (for now) :)

Reply:
That makes sense. I suppose if two mixins override `modelIsVisible`, then whichever mixin is included last in the list is the one that sticks?

Reply:
@pushmatrix Actually, our mixins use inheritance to layer onto the class. So, as long as a mixin invokes the accessor on its super class, it can retain the implementation of other mixins without knowing what that implementation does. In this case, it would look like this in `LoadingMixin`:

```typescript
get modelIsVisible(): boolean {
  return super.modelIsVisible && this[$posterHidden];
}
```